### PR TITLE
Add fren.ai to cosmos/chain-registry

### PR DESCRIPTION
**Pull request by cosmosscanner**
## Summary

This pull request adds the initial registry information for the `fren.ai` chain to the `cosmos/chain-registry` repository.

## Changes

- Added `frenai/assetlist.json` with the following details:
  - Description: The native token of frenai
  - Denom Units: ufren (base unit), fren (display unit)
  - Type: sdk.coin
  - Symbol: FREN
  - Logo URIs and images
  - Keywords: ai, ai data, ai polling, rpc, ai rpc, ai grpc, ai rest api, ai chatbot, ai processed data
  - Socials: website, X/Twitter

- Added `frenai/chain.json` with the following details:
  - Chain ID: fren-1
  - Bech32 Prefix: fren
  - Denom: ufren
  - Gas Prices: fixed, low, average, high
  - Explorer URL and transaction page
  - RPC, REST, and gRPC endpoints
  - Seed nodes and peer nodes

## Notes

- The images for the token logo are included in the `frenai/images` directory.
- Please review the added details and let me know if there are any adjustments needed.

Thank you for considering this addition to the `cosmos/chain-registry` repository.

### CODE REVIEW

File: frenai/chain.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "chain_id": "fren-1",
+  "website": "https://fren.ai",
+  "pretty_name": "fren.ai",
+  "bech32_prefix": "fren",
+  "denom": "ufren",
+  "fixed_min_gas_price": 1e-9,
+  "low_gas_price": 1e-9,
+  "average_gas_price": 0.00005,
+  "high_gas_price": 0.0001,
+  "explorer": {
```
Comment (JeremyParish69) [inline frenai/chain.json:13]: explorers in an array, and this object should go inside that array

File: frenai/chain.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "url": "https://explorer.fren.ai",
+    "tx_page": "/transactions/{txHash}"
+  },
+  "rpc": [
```
Comment (JeremyParish69) [inline frenai/chain.json:17]: rpc, rest, grpc should go inside the api object

File: frenai/assetlist.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          "denom": "fren",
+          "exponent": 6
+        }
+      ],
+      "type_asset": "sdk.coin",
+      "base": "ufren",
+      "name": "frenai",
+      "display": "fren",
+      "symbol": "FREN",
+      "logo_URIs": {
```
Comment (JeremyParish69) [inline frenai/assetlist.json:22]: Please remove logo_URIs

File: .github/workflows/python-publish.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package
```
Comment (JeremyParish69) [inline .github/workflows/python-publish.yml:9]: Why add this? Please remove

File: frenai/chain.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
+  "$schema": "../chain.schema.json",
+  "chain_name": "frenai",
+  "chain_id": "fren-1",
+  "website": "https://fren.ai",
+  "pretty_name": "fren.ai",
+  "bech32_prefix": "fren",
```
[the remark this answers is not in the document]
Comment (cosmosscanner) [reply frenai/chain.json:7]: Pull request `Add Fren.ai to slip-0173.md` sent

File: frenai/chain.json [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      }
+    ],
+    "persistent_peers": [
+      {
+        "address": "98.76.54.32:26656",
+        "provider": "fren.ai"
+      }
+    ]
+  },
+  "logo_URIs": {
```
Comment (JeremyParish69) [inline frenai/chain.json:91]: Please use `images` instead of `logo_URIs`

File: frenai/chain.json [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    "genesis": {
+      "genesis_url": "https://raw.githubusercontent.com/frenailabs/networks/main/fren-1/genesis.json"
+    }
+  },
+  "explorers": [
+    {
+      "kind": "explorer",
+      "url": "https://explorer.fren.ai",
+      "tx_page": "/transactions/{txHash}"
```
Comment (JeremyParish69) [inline frenai/chain.json:54]: Please include use the full URL

File: frenai/assetlist.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        "ai grpc",
+        "ai rest api",
+        "ai chatbot",
+        "ai processed data"
+      ],
+      "socials": {
+        "website": "https://fren.ai",
+        "twitter": "https://twitter.com/frenaihub"
+      }
+    }
```
Comment (JeremyParish69) [inline frenai/assetlist.json:37]: No `images` for $FREN's logo?

File: frenai/chain.json [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    "staking_tokens": [
+      {
+        "denom": "ufren"
+      }
+    ]
+  },
+  "codebase": {
+    "git_repo": "https://github.com/frenailabs/frenai/",
+    "recommended_version": "v1.0.0",
+    "compatible_versions": ["v1.0.0"],
```
Comment (JeremyParish69) [inline frenai/chain.json:35]: Please duplicate this version data into a version history structure (either a `versions` array in this file or `versions.json` file.

File: frenai/chain.json [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+  "codebase": {
+    "git_repo": "https://github.com/frenailabs/frenai/",
+    "recommended_version": "v0.1.0",
+    "compatible_versions": ["v0.1.0"],
+    "binaries": {
+      "linux/amd64": "https://github.com/frenailabs/releases/download/v0.1.0/frenaid-v0.1.0-linux-amd64.tar.gz",
+      "linux/arm64": "https://github.com/frenailabs/releases/download/v0.1.0/frenaid-v0.1.0-linux-arm64.tar.gz"
+    },
+    "cosmos_sdk_version": "v0.50.6",
+    "ibc_go_version": "v8.0.0",
```
Comment (JeremyParish69) [inline frenai/chain.json:41]: Can you please add the cosmos_sdk_verions and the ibc_go_version into the versions array as well?

File: frenai/assetlist.json [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        "ai rest api",
+        "ai chatbot",
+        "ai processed data"
+      ],
+      "socials": {
+        "website": "https://fren.ai",
+        "twitter": "https://twitter.com/frenaihub"
+      },
+      "images": {
+        "png": "https://raw.githubusercontent.com/frenailabs/chain-registry/master/frenai/images/fren.png",
```
Comment (JeremyParish69) [inline frenai/assetlist.json:38]: These URIs must point to the Cosmos Chain Registry. I think simply replacing 'frenailabs' org with 'cosmos' should do the trick.

File: frenai/chain.json [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+      {
+        "address": "https://api.fren.ai:9090",
+        "provider": "fren.ai"
+      }
+    ]
+  },
+  "peers": {
+    "seeds": [
+      {
+        "address": "123.45.67.89:26656",
```
Comment (JeremyParish69) [inline frenai/chain.json:94]: `id` is a required property for all peers

File: frenai/chain.json (hunk 10 above)
Comment (cosmosscanner) [reply frenai/chain.json:41]: I've added the `cosmos_sdk_version` and `ibc_go_version` to the `versions` array as requested. Thank you for the feedback.

File: frenai/assetlist.json (hunk 11 above)
Comment (cosmosscanner) [reply frenai/assetlist.json]: Replaced.

File: frenai/chain.json (hunk 12 above)
Comment (cosmosscanner) [reply frenai/chain.json:97]: Added placeholder  "id": "TBD",

File: frenai/chain.json [hunk 13]
```diff
@@ -0,0 +1,116 @@
+{
+  "$schema": "../chain.schema.json",
+  "chain_name": "frenai",
+  "status": "development",
```
Comment (JeremyParish69) [inline frenai/chain.json:4]: This is not a valid status. See:
https://github.com/cosmos/chain-registry/blob/099474d5c49ed2dc7ba62a68b96e3f3c96e7284d/chain.schema.json#L36

File: frenai/chain.json (hunk 10 above)
Comment (JeremyParish69) [reply frenai/chain.json:41]: Sorry, and `consensus`, too, please. (should include all the codebase data that's not git_repo and genesis)

File: frenai/chain.json [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+        "address": "https://dexfx.ai/grpc",
+        "provider": "fren.ai"
+      }
+    ]
+  },
+  "peers": {
+    "seeds": [
+      {
+        "id": "TBD",
+        "address": ":26656",
```
Comment (JeremyParish69) [inline frenai/chain.json:97]: Should we be waiting for more info on these? or were they added as placeholders? If placeholders, please remove them, and if the arrays would be empty, then remove the arrays entirely, as well.

**Comment by JeremyParish69** [comment]
Is there still intent to complete this?

**Pull request closed without merging**